src/Ledger/Utxo/Properties.lagda

```diff
@@ -544,3 +548,194 @@ pov {tx}{utxo}{_}{fees}{deposits}{donations} h'
 \end{code}
 
 \end{property}
+
+\begin{property}[\textbf{General Minimum Spending Condition}]~\\
+
+\begin{code}[hide]
+isRefundCert : DCert → Bool
+isRefundCert (dereg c) = true
+isRefundCert (deregdrep c) = true
+isRefundCert _ = false
+
+noRefundCert : List DCert → Set _
+noRefundCert l = All (λ cert → isRefundCert cert ≡ false) l
+
+opaque
+  unfolding List-Model
+  unfolding finiteness
+  fin∘list[] : {A : Set} → proj₁ (finiteness{A = A} ∅) ≡ []
+  fin∘list[] = refl
+  fin∘list∷[] : {A : Set} {a : A} → proj₁ (finiteness ❴ a ❵) ≡ [ a ]
+  fin∘list∷[] = refl
+
+coin∅ : getCoin{A = DepositPurpose ⇀ Coin} ∅ ≡ 0
+coin∅ = begin
+  foldr (λ x → (proj₂ x) +_) 0 (deduplicate _≟_ (proj₁ (finiteness ∅)))
+    ≡⟨ cong (λ u → (foldr (λ x → (proj₂ x) +_) 0 (deduplicate _≟_ u))) fin∘list[] ⟩
+  foldr (λ (x : DepositPurpose × Coin) → (proj₂ x) +_) 0 (deduplicate _≟_ [])
+    ≡⟨ cong (λ u → (foldr (λ (x : DepositPurpose × Coin) → (proj₂ x) +_) 0  u))
+            {x = deduplicate _≟_ []} {y = []} refl ⟩
+  foldr (λ (x : DepositPurpose × Coin) → (proj₂ x) +_) 0 []
+    ≡⟨ refl ⟩
+  0 ∎
+  where open Prelude.≡-Reasoning
+
+getCoin-singleton : ((dp , c) : DepositPurpose × Coin) → indexedSumᵛ' id ❴ (dp , c) ❵ ≡ c
+getCoin-singleton (dp , c) = begin
```

Comment:
Here's an alternative proof:
```
getCoin-singleton _ = indexedSum-singleton' ⦃ M = +-0-commutativeMonoid ⦄ (finiteness _)
```
The same could be done for `coin∅`, but there's only `indexedSum-∅`, not a `indexedSum-∅'` variant that works with an arbitrary finiteness proof. The proof for `indexedSum-∅'` would be the same as for `indexedSum-singleton'` if you want to do it.

Reply:
Nice! 🚀

Reply:
I'll revisit this and possibly implement `indexedSum-∅'` once everything else is done.